sklearn/svm/src/libsvm/svm.cpp

```diff
@@ -1423,17 +1467,38 @@ class SVC_Q: public Kernel
 		QD = new double[prob.l];
 		for(int i=0;i<prob.l;i++)
 			QD[i] = (this->*kernel_function)(i,i);
+#ifdef _DENSE_REP
+		this->shrinking = param.shrinking;
+		this->l1 = prob.l1;
+		this->l2 = prob.l2;
+		this->kernel_type = param.kernel_type;
+#endif
 	}
 	
 	Qfloat *get_Q(int i, int len) const
 	{
 		Qfloat *data;
 		int start, j;
-		if((start = cache->get_data(i,&data,len)) < len)
-		{
-			for(j=start;j<len;j++)
-				data[j] = (Qfloat)(y[i]*y[j]*(this->*kernel_function)(i,j));
+
+		start = cache->get_data(i, &data, len);
+#ifdef _DENSE_REP
+		if (shrinking==0 && kernel_type==RBF) {
```

Comment:
My concern with this PR is that it adds a fair bit of complexity to speeding up RBF + no shrinking.